From 36af72c2d525f62bf7efb11ba242ee628817935a Mon Sep 17 00:00:00 2001
From: Stefan Lange <s.lange@gateware.de>
Date: Wed, 7 Mar 2018 16:42:27 +0100
Subject: [PATCH] TQMT2081: dts: add xfi_tx_dis gpio hog, extend i2c commentary

Add GPIO hog to automatically unset XFI_TX_DIS at boot time, enabling
the SFP transceiver's tx output.

Extend commentary describing usage of I2C GPIO expanders.

Signed-off-by: Stefan Lange <s.lange@gateware.de>
---
 arch/powerpc/boot/dts/fsl/tqmt2081.dts | 17 +++++++++++++----
 1 file changed, 13 insertions(+), 4 deletions(-)

diff --git a/arch/powerpc/boot/dts/fsl/tqmt2081.dts b/arch/powerpc/boot/dts/fsl/tqmt2081.dts
index d217ec3..69fe14c 100644
--- a/arch/powerpc/boot/dts/fsl/tqmt2081.dts
+++ b/arch/powerpc/boot/dts/fsl/tqmt2081.dts
@@ -198,25 +198,34 @@ reserved-memory {
 				compatible = "nxp,pca9530";
 				reg = <0x60>;
 			};
-			/* PCA9554 GPIO: DIP Switches */
+			/* PCA9554 GPIO (D56): Buttons, GPIO DIP Switches */
 			gpio@20 {
 				compatible = "nxp,pca9554";
 				reg = <0x20>;
 			};
-			/* PCA9554 GPIO: PCIe & LCD control */
+			/* PCA9554 GPIO (D55): PCIe & LCD control */
 			gpio@21 {
 				compatible = "nxp,pca9554";
 				reg = <0x21>;
 			};
-			/* PCA9554 GPIO: SerDes & DIU Mux, LEDs*/
+			/* PCA9554 GPIO (D58): SerDes Mux DIP Switches, LEDs*/
 			gpio@22 {
 				compatible = "nxp,pca9554";
 				reg = <0x22>;
 			};
-			/* PCA9554 GPIO: XFI control, SGMII INT# */
+			/* PCA9554 GPIO (D57): XFI control, ETH INT# */
 			gpio@23 {
 				compatible = "nxp,pca9554";
 				reg = <0x23>;
+				gpio-controller;
+				#gpio-cells = <0x2>;
+
+				xfi_tx_dis {
+					gpio-hog;
+					gpios = <1 0>;
+					output-low;
+					line-name = "xfi_tx_dis-gpio";
+				};
 			};
 			/* STMPE touchscreen controller */
 			stmpe811: stmpe811@41 {
-- 
1.9.1

